components/raftstore/src/store/peer.rs

```diff
@@ -3344,6 +3332,87 @@ where
         assert!(self.pending_merge_state.is_some());
         self.want_rollback_merge_peers.insert(peer_id);
     }
+
+    // Check disk usages for the peer itself and other peers in the raft group.
+    // The return value indicates whether the propose is allowed or not.
+    fn check_disk_usages_before_propose_normal<T>(
+        &mut self,
+        ctx: &mut PollContext<EK, ER, T>,
+        disk_full_opt: DiskFullOpt,
+        disk_full_stores: &mut Vec<u64>,
+    ) -> bool {
+        // Propose can be rejected if of the store's disk is full.
+        let self_allowed = match disk_full_opt {
+            DiskFullOpt::NotAllowedOnFull => matches!(ctx.self_disk_usage, DiskUsage::Normal),
+            _ => !matches!(ctx.self_disk_usage, DiskUsage::AlreadyFull),
```

Comment:
If it's `AllowedOnAlreadyFull`, is it can be proposed?

Reply:
`AllowedOnAlreadyFull` is not used exactly. We will remove it from `kvproto` later.